pkg/ossm/operator.go

```diff
@@ -0,0 +1,139 @@
+	if err != nil {
+		util.Log.Error("Failed to taint the infra node")
+	}
+	util.Log.Info(response)
+	if strings.Contains(response, "untainted") {
+		util.Log.Info("Infra node is not tainted anymore")
+	} else {
+		util.Log.Error("Fail to remove the taint from the infra node")
+	}
+
+}
+
+// TestOperator tests scenario to cover all the test cases related to the OSSM operators
+func TestOperator(t *testing.T) {
+	defer cleanupOperatorTest()
+	defer util.RecoverPanic(t)
+
+	util.Log.Info("Test cases related to OSSM Operators")
+
+	//test case regarding the https://issues.redhat.com/browse/OSSM-2342 issue to run OSSM operator on infra nodes
+	t.Run("test_ossm_operator_deploy_infra_nodes", func(t *testing.T) {
+		defer util.RecoverPanic(t)
+		util.Log.Info("Testing: Run OSSM Operator on infra nodes")
+		//Get and pick one worker node that does not have already installed the istio operator
+		workername := pickWorkerNode(t)
+		//Label the worker node as infra
+		_, err := util.Shell(`oc label node %s node-role.kubernetes.io/infra=`, workername)
+		if err != nil {
+			t.Fatalf("Failed to label the worker node as infra")
+		}
+		_, err = util.Shell(`oc label node %s node-role.kubernetes.io=infra`, workername)
+		if err != nil {
+			t.Fatalf("Failed to label the worker node as infra")
+		}
+		//verify that the worker node is labeled as infra
+		name, err := util.Shell(`oc get nodes -l node-role.kubernetes.io/infra= -o jsonpath='{.items[0].metadata.name}'`)
+		if err != nil {
+			t.Fatalf("Failed to get the infra node name")
+		}
+		if name != workername {
+			t.Fatalf("Failed to label the worker node as infra")
+		}
+		//Taint the node. The only validation to check this is the output message
+		response, err := util.Shell(`oc adm taint nodes -l node-role.kubernetes.io/infra node-role.kubernetes.io/infra=reserved:NoSchedule node-role.kubernetes.io/infra=reserved:NoExecute`)
+		if err != nil {
+			t.Fatalf("Failed to taint the infra node")
+		}
+		if !strings.Contains(response, "tainted") {
+			t.Fatalf("Failed to taint the infra node")
+		}
+		//Edit the subscription to add the infra node to the node selector
+		_, err = util.Shell(`oc patch subscription servicemeshoperator -n openshift-operators --type merge -p '{"spec":{"config":{"nodeSelector":{"node-role.kubernetes.io/infra":""},"tolerations":[{"effect":"NoSchedule","key":"node-role.kubernetes.io/infra","value":"reserved"},{"effect":"NoExecute","key":"node-role.kubernetes.io/infra","value":"reserved"}]}}}'`)
+
+		if err != nil {
+			t.Fatalf("Failed to patch the subscription")
+		}
+		//Verify that the operator pod is running on the infra node
+		podname, err := util.Shell(`oc get pods -n openshift-operators -l name=istio-operator --field-selector spec.nodeName=%s -o jsonpath='{.items[0].metadata.name}'`, workername)
+		if err != nil {
+			time.Sleep(time.Duration(20) * time.Second)
```

Comment:
@fjglira Is there a better way to check operator pod in running state before checking its pod name ?
I see you are using hard coded 20s sleep time here. Is that enough for an operator pod to run after running the oc patch command above ?

Reply:
Modified this to make retries to avoid time.sleep